src/WebJobs.Script/Rpc/ProcessManagement/IProcessRegistry.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace Microsoft.Azure.WebJobs.Script.Rpc
 {
```

Comment:
Lots of types having their visibility changed to public. Is that a requirement? Why?

Reply:
needed to make FunctionRegistrationContext public as registration of functions is not decoupled with creating a language worker and languageworkerchannelmanager needs access. Reverted any unintended types I accidentally made public

Reply:
@mathewc - Added IFunctionDispatcher as a dependency to ScriptHost --> This required making IFunctionDispatcher  public --> This required making FunctionRegistrationContext public